### PR TITLE
Set Transition At Correctly

### PR DESCRIPTION
**Pull request by gsnovak**
Hopefully I won't break everything again, but @fiestacasey did some fancy figuring and it was probably the fact the we don't always return a boolean for before transitions.

### CODE REVIEW

File: lib/ar_state_machine.rb [hunk 1]
```diff
@@ -83,7 +83,7 @@ def save_state_change
   end
 
   def do_state_change_before_callbacks
-    return false unless self.class.run_before_transition_callbacks(self.state, self, old_state)
+    return false if self.class.run_before_transition_callbacks(self.state, self, old_state) == false
```
Comment (ksolkowski) [inline lib/ar_state_machine.rb]: does it ever return nil?

Comment (gsnovak) [reply lib/ar_state_machine.rb]: yea, that was actually the issue we were having.

File: lib/ar_state_machine/version.rb [hunk 2]
```diff
@@ -1,3 +1,3 @@
 module ARStateMachine
-  VERSION = "0.1.6"
+  VERSION = "0.1.8"
```
Comment (fiestacasey) [inline lib/ar_state_machine/version.rb]: needs bundle install

Comment (gsnovak) [reply lib/ar_state_machine/version.rb]: i think i did that right before you said dis

**Pull request merged**